src/plugins/anki/anki-plugin.tsx

```diff
@@ -0,0 +1,291 @@
+        setReachable(false);
+      })
+      .then((data: number[]) => {
+        if (reachable) {
+          setCardsToLearn(data);
+          getCardAndSet(data[0]);
+        } else {
+          setCardsToLearn([]);
+        }
+      });
+  };
+
+  useEffect(init, []);
+
+  const again = () => {
+    invoke("answerCards", 6, {
+      answers: [
+        {
+          cardId: cardsToLearn[0],
+          ease: 1,
+        },
+      ],
+    }).then((data) => {
+      if (cardsToLearn.length == 1) {
+        init();
+        return;
+      }
+      getCardAndSet(cardsToLearn[1]);
+      setCardsToLearn(cardsToLearn.slice(1));
+      setDisplay("question");
+    });
+  };
+
+  const easy = () => {
+    invoke("answerCards", 6, {
+      answers: [
+        {
+          cardId: cardsToLearn[0],
+          ease: 4,
+        },
+      ],
+    }).then((data) => {
+      if (cardsToLearn.length == 1) {
+        init();
+        return;
+      }
+      getCardAndSet(cardsToLearn[1]);
+      setCardsToLearn(cardsToLearn.slice(1));
+      setDisplay("question");
+    });
+  };
+
+  return (
+    <div className="AnkiWidget">
+      <p
+        className="setNameLine"
+        style={{ display: reachable ? "block" : "none" }}
+      >
+        <span className="setName">{config?.deckName}</span>
+        <span>(Due: {cardsToLearn?.length})</span>
+      </p>
+
+      <div style={{ display: reachable ? "block" : "none" }} className="card">
+        <div
+          className="card_question_wrap"
+          dangerouslySetInnerHTML={currentCard[display]}
+        ></div>
+      </div>
+
+      <div style={{ display: reachable ? "none" : "block" }}>
+        <p>{t("anki-plugin.error")}</p>
+      </div>
+
+      <div className="actions" style={{ display: reachable ? "flex" : "none" }}>
+        <Button
+          style={{ display: display === "question" ? "block" : "none" }}
+          onClick={() => {
+            if (display === "question") {
+              setDisplay("answer");
+            } else {
+              setDisplay("question");
+            }
+          }}
+        >
+          {t("anki-plugin.showAnswer")}
+        </Button>
+
+        <Button
+          style={{ display: display === "question" ? "none" : "block" }}
+          onClick={easy}
+        >
+          {t("anki-plugin.easy")}
+        </Button>
+        <Button
+          style={{ display: display === "question" ? "none" : "block" }}
+          onClick={again}
+        >
+          {t("anki-plugin.again")}
+        </Button>
+      </div>
+    </div>
+  );
+};
+
+const onCommandInput: OnCommandInputCallback = async (text: string) => {
+  const q = text.toLowerCase();
+  const widgets = await getAllWidgetsByPlugin(ankiPlugin);
+
+  return [];
+};
+
+const widgetDescriptor = {
+  id: "widget",
+  get name() {
+    return translate("anki-plugin.widgetName");
+  },
+  configurationScreen: WidgetConfigScreen,
+  mainScreen: MainScreen,
+  mock: () => {
+    return (
+      <MainScreen
```

Comment:
not quite there will be the error message

Reply:
Could you update mock to look like a real widget?

Reply:
it already uses the default set of anki, another set would be quite troublesome, since we don't know which set the user has any active cards in it ...

Reply:
I'm not sure I understand what you mean. What I initially meant is to make mock widget look like legit one, it doesn't need to be working per se or have real data. All values can (and should) be hardcoded, so it will work without Anko/AnkiConnect installed. You can create any example card to show there

Reply:
ah okay, since im currently rather busy, i will do it in the coming week